src/amberscript/parser.cc

```diff
@@ -1309,5 +1326,50 @@ Result Parser::ParseDeviceFeature() {
   return ValidateEndOfStatement("DEVICE_FEATURE command");
 }
 
+Result Parser::ParseRepeat() {
```

Comment:
This whole thing works without a stack because REPEAT is not itself repeatable.  Works for me.